### PR TITLE
Delete message feature

### PR DESCRIPTION
**Pull request by shahin1290**
## Pivotal Tracker link
[Link to PT](https://www.pivotaltracker.com/story/show/162084276)

## Actions taken
Created A feature file -> delete_message.feature
Added feature, user story, scenarios, steps

## What we have learned
How to create  and set up factory bot, mail boxer gem, using Warden to log in user 
And lots of git ponging

### CODE REVIEW

File: features/step_definitions/basic_steps.rb [hunk 1]
```diff
@@ -27,3 +29,33 @@
...
+Given(/^I am logged in as "([^"]*)"$/) do |name|
+  @user = User.find_by(name: name)
+  login_as(@user, scope: :user)
+end
+
+Given("I am on the inbox page") do
+    visit mailbox_inbox_path
+end
+
+Given(/^I send a mail to "([^"]*)"$/) do |name|
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: Why are you using regex?

File: features/step_definitions/basic_steps.rb [hunk 2]
```diff
@@ -27,3 +29,33 @@
...
+    @receiver = User.find_by(name: name)
+    @user.send_message(@receiver, 'Lorem ipsum...', 'Subject')
+end
+
+Given("I am on the home page") do
+    visit root_path
+end
+
+Then(/^I should have "([^"]*)" messages$/) do |expected_count|
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: same here regex

File: features/step_definitions/basic_steps.rb [hunk 3]
```diff
@@ -27,3 +29,33 @@
     save_and_open_page
 end
 
+# Delete Message feature
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: No need for comments

File: features/step_definitions/basic_steps.rb [hunk 4]
```diff
@@ -1,4 +1,6 @@
 
+# Sign Up feature
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: no need for comments

File: features/sign_up_page.feature [hunk 5]
```diff
@@ -9,24 +9,24 @@ Scenario: Submit sign up form [Happy path]
...
     Then I fill in "Email" with " "
-    When I click "Create" to submit form
+    When I click "Create" 
     Then I should see "Email can't be blank"
  
 Scenario: Submit sign up form without password [Sad path]
     When I visit the "sign up" page
     Then I fill in "Password" with " "
-    When I click "Create" to submit form
+    When I click "Create"
```
Comment (faraznaeem) [inline features/sign_up_page.feature]: Why are you adding the `"Create"` in the signup feature file?
Please explain the reason here.

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: features/step_definitions/basic_steps.rb (hunk 1 above)
Comment (shahin1290) [reply features/step_definitions/basic_steps.rb]: No, we do not need it and replaced it

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (shahin1290) [reply features/step_definitions/basic_steps.rb]: No, we do not need it and replaced it

File: features/step_definitions/basic_steps.rb (hunk 3 above)
Comment (shahin1290) [reply features/step_definitions/basic_steps.rb]: No, we do not need it and deleted it

File: features/step_definitions/basic_steps.rb (hunk 4 above)
Comment (shahin1290) [reply features/step_definitions/basic_steps.rb]: No, we do not need it and deleted it

File: features/sign_up_page.feature (hunk 5 above)
Comment (shahin1290) [reply features/sign_up_page.feature]: 1. We did not add but modified that line
2. The submit button on the sign up page is called Create

File: features/delete_message.feature [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    | name   | email             | password | password_confirmation |
+    | Jenny  | jenny@ranom.com   | password | password              |
+    | Daniel | daniel@random.com | password | password              |
+
+Scenario: Deleting a message
+  Given I am logged in as "Daniel"
+  And I am on the inbox page
+  And I send a mail to "Jenny"
+  And I click "Logout" 
+  Given I am logged in as "Jenny"
```
Comment (magnus-thor) [inline features/delete_message.feature:18]: The steps above this one can and should be done in the background using a table
```
Given the following email exists
| sender | receiver | body          | subject           |
| Daniel  | Jenny    | some message | some subject  |
```

File: features/sign_up_page.feature [hunk 7]
```diff
@@ -9,24 +9,24 @@ Scenario: Submit sign up form [Happy path]
...
     And I fill in "Password confirmation" with "camron1234"
-    When I click "Create" to submit form
+    When I click "Create" 
     And I should see "Welcome! You have signed up successfully."
 
 Scenario: Submit sign up form without name [Sad path]
     When I visit the "sign up" page
     Then I fill in "Name" with " "    
-    When I click "Create" to submit form
+    When I click "Create"
```
Comment (magnus-thor) [inline features/sign_up_page.feature]: not a blocker for this pr but when you write the scenario try to follow this pattern
```
- (Given) some context
- (When) some action is carried out
- (Then) a particular set of observable consequences should obtain
```

**Review by magnus-thor — CHANGES_REQUESTED** [review]

File: features/sign_up_page.feature (hunk 7 above)
Comment (CamronLDNF) [reply features/sign_up_page.feature]: Noted. Thanks.

File: features/reply_to_message.feature [hunk 8]
```diff
@@ -0,0 +1,28 @@
+Feature: Reply to message
+    As a user
+    In order to send a response
+    I want to be able to reply to a message in my inbox
+
+Scenario: Submit sign up form [Happy path]
+- (Given) some context
+- (When) some action is carried out
+- (Then) a particular set of observable consequences should obtain
```
Comment (faraznaeem) [inline features/reply_to_message.feature]: Follow the standard here `-(Given)` is not needed just do `Given`

Comment (shahin1290) [reply features/reply_to_message.feature]: removed

File: features/delete_message.feature (hunk 6 above)
Comment (shahin1290) [reply features/delete_message.feature:18]: done

File: features/sign_up_page.feature (hunk 5 above)
Comment (shahin1290) [reply features/sign_up_page.feature]: done

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**